### PR TITLE
Fix missing not in format._put_lines

### PR DESCRIPTION
**Pull request by limx0**

### CODE REVIEW

File: pandas/formats/format.py [hunk 1]
```diff
@@ -2715,7 +2715,7 @@ def set_eng_float_format(accuracy=3, use_eng_prefix=False):
 
 
 def _put_lines(buf, lines):
-    if any(isinstance(x, compat.text_type) for x in lines):
+    if any(not isinstance(x, compat.text_type) for x in lines):
```
Comment (jreback) [inline pandas/formats/format.py]: why are you suggesting this change? This changes the display output of ``.info()``. what exactly is the problem?

**Comment by limx0** [comment]
@jreback the following raises a TypeError

```python
import pandas as pd
from io import StringIO
buf = StringIO()
pd.DataFrame().info(buf=buf)

   2642     if any(isinstance(x, compat.text_type) for x in lines):
   2643         lines = [compat.text_type(x) for x in lines]
-> 2644     buf.write('\n'.join(lines))
   2645 
   2646 

TypeError: unicode argument expected, got 'str'
````
 Which I believe this PR fixes (and is the proper intention of the code)

**Comment by jorisvandenbossche** [comment]
@limx0 Can you add a test for the actual `info` issue?

File: pandas/tests/formats/test_format.py [hunk 2]
```diff
@@ -2694,6 +2694,13 @@ def test_float_trim_zeros(self):
                 self.assertTrue(('+10' in line) or skip)
             skip = False
 
+    def test_put_lines(self):
+        from io import StringIO
```
Comment (jorisvandenbossche) [inline pandas/tests/formats/test_format.py]: StringIO is already imported from pandas.compat

Comment (jorisvandenbossche) [inline pandas/tests/formats/test_format.py]: further, can you add the pr number in a comment on the first line?

File: pandas/tests/frame/test_repr_info.py [hunk 3]
```diff
@@ -381,3 +381,10 @@ def test_info_memory_usage(self):
         # deep=True, and add on some GC overhead
         diff = df.memory_usage(deep=True).sum() - sys.getsizeof(df)
         self.assertTrue(abs(diff) < 100)
+
+    def test_info_empty_frame(self):
+        io = StringIO()
+        pd.DataFrame().info(buf=io)
+        assert (io.getvalue() == "<class 'pandas.core.frame.DataFrame'>\n"
+                                 "Index: 0 entries\n"
+                                 "Empty DataFrame")
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/test_repr_info.py]: This test already works for me with latest master (and latest release). So I don't think this is testing what you changed in the code

**Comment by limx0** [comment]
@jorisvandenbossche this is now working in the latest master. Closing this PR

**Pull request closed without merging**